src/function/cast/varint_casts.cpp

```diff
@@ -0,0 +1,533 @@
+				// we can cap this
+				digits.pop_back();
+			}
+			digit_idx--;
+		}
+		if (is_negative) {
+			blob_string.push_back(static_cast<char>(~remainder));
+		} else {
+			blob_string.push_back(static_cast<char>(remainder));
+		}
+	}
+
+	uint32_t blob_size = static_cast<uint32_t>(blob_string.size() + VARINT_HEADER_SIZE);
+	result_value = StringVector::EmptyString(result, blob_size);
+	auto writable_blob = result_value.GetDataWriteable();
+
+	SetHeader(writable_blob, blob_string.size(), is_negative);
+
+	// Write string_blob into blob
+	idx_t blob_string_idx = blob_string.size() - 1;
+	for (idx_t i = VARINT_HEADER_SIZE; i < blob_size; i++) {
+		writable_blob[i] = blob_string[blob_string_idx--];
+	}
+	result_value.Finalize();
+	return true;
+}
+
+template <class T>
+bool DoubleToVarInt(T double_value, string_t &result_value, Vector &result) {
+	// Check if we can cast it
+	if (!std::isfinite(double_value)) {
+		// We can't cast inf -inf nan
+		return false;
+	}
+	// Determine if the number is negative
+	bool is_negative = double_value < 0;
+	// Determine the number of data bytes
+	double abs_value = std::abs(double_value);
+
+	if (abs_value == 0) {
+		// Return Value 0
+		result_value = ZeroBlob(result);
+		return true;
+	}
+	vector<char> value;
+	while (abs_value > 0) {
+		double quotient = abs_value / 256;
+		double truncated = floor(quotient);
+		uint8_t byte = static_cast<uint8_t>(abs_value - truncated * 256);
+		abs_value = truncated;
+		if (is_negative) {
+			value.push_back(~static_cast<char>(byte));
+		} else {
+			value.push_back(static_cast<char>(byte));
+		}
+	}
+	uint32_t data_byte_size = static_cast<uint32_t>(value.size());
+	uint32_t blob_size = data_byte_size + VARINT_HEADER_SIZE;
+	result_value = StringVector::EmptyString(result, blob_size);
+	auto writable_blob = result_value.GetDataWriteable();
+	SetHeader(writable_blob, data_byte_size, is_negative);
+	// Add data bytes to the blob, starting off after header bytes
+	idx_t blob_string_idx = value.size() - 1;
+	for (idx_t i = VARINT_HEADER_SIZE; i < blob_size; i++) {
+		writable_blob[i] = value[blob_string_idx--];
+	}
+	result_value.Finalize();
+	return true;
+}
+
+template <>
+bool TryCastToVarInt::Operation(double double_value, string_t &result_value, Vector &result,
+                                CastParameters &parameters) {
+	return DoubleToVarInt(double_value, result_value, result);
+}
+
+template <>
+bool TryCastToVarInt::Operation(float double_value, string_t &result_value, Vector &result,
+                                CastParameters &parameters) {
+	return DoubleToVarInt(double_value, result_value, result);
+}
+
+struct VarIntCastToVarchar {
+	template <class SRC>
+	static inline string_t Operation(SRC input, Vector &result) {
+		return StringVector::AddStringOrBlob(result, VarIntToVarchar(input));
+	}
+};
+
+bool VarintToDouble(string_t &blob, double &result, bool &strict) {
+	result = 0;
+	bool is_negative;
```

Comment:
Maybe we can add a `Varint` class similar to `Hugeint` and `Timestamp` that holds all of these methods?